### PR TITLE
Use SafeLoader instead of FullLoader

### PR DESCRIPTION
**Pull request by neoaggelos**
## Summary

This minor PR is to make use of `yaml.safe_load` when loading YAML files in Python. This is to prevent issues (security or otherwise) that may arise from using the `yaml.FullLoader`

### CODE REVIEW

File: scripts/cluster/common/utils.py [hunk 1]
```diff
@@ -126,7 +126,7 @@ def get_dqlite_port():
     port = 19001
     if os.path.exists(dqlite_info):
         with open(dqlite_info) as f:
-            data = yaml.load(f, Loader=yaml.FullLoader)
+            data = yaml.safe_load(f)
```
Comment (ktsakalozos) [inline scripts/cluster/common/utils.py:129]: Trying this, I got this error:
```
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]: Waiting for access to cluster. [2021-11-18 10:30:50,362] ERROR in app: Exception on /cluster/api/v2.0/join [POST]
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]: Traceback (most recent call last):
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:   File "/snap/microk8s/x1/lib/python3.6/site-packages/flask/app.py", line 2447, in wsgi_app
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:     response = self.full_dispatch_request()
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:   File "/snap/microk8s/x1/lib/python3.6/site-packages/flask/app.py", line 1952, in full_dispatch_request
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:     rv = self.handle_user_exception(e)
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:   File "/snap/microk8s/x1/lib/python3.6/site-packages/flask/app.py", line 1821, in handle_user_exception
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:     reraise(exc_type, exc_value, tb)
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:   File "/snap/microk8s/x1/lib/python3.6/site-packages/flask/_compat.py", line 39, in reraise
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:     raise value
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:   File "/snap/microk8s/x1/lib/python3.6/site-packages/flask/app.py", line 1950, in full_dispatch_request
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:     rv = self.dispatch_request()
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:   File "/snap/microk8s/x1/lib/python3.6/site-packages/flask/app.py", line 1936, in dispatch_request
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:     return self.view_functions[rule.endpoint](**req.view_args)
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:   File "/snap/microk8s/x1/scripts/cluster/agent.py", line 600, in join_node_dqlite
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:     voters, non_voters = get_dqlite_nodes()
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:   File "/snap/microk8s/x1/scripts/cluster/agent.py", line 469, in get_dqlite_nodes
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:     data = data.safe_load(f)
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]: UnboundLocalError: local variable 'data' referenced before assignment
```

**Review by ktsakalozos — CHANGES_REQUESTED** [review]
Thank you for this PR @neoaggelos

File: scripts/cluster/agent.py [hunk 2]
```diff
@@ -466,7 +466,7 @@ def get_dqlite_nodes():
     while waits > 0:
         try:
             with open("{}/info.yaml".format(cluster_dir)) as f:
-                data = yaml.load(f, Loader=yaml.FullLoader)
+                data = data.safe_load(f)
```
Comment (ktsakalozos) [inline scripts/cluster/agent.py:469]: Trying this, I got this error:
```
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]: Waiting for access to cluster. [2021-11-18 10:30:50,362] ERROR in app: Exception on /cluster/api/v2.0/join [POST]
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]: Traceback (most recent call last):
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:   File "/snap/microk8s/x1/lib/python3.6/site-packages/flask/app.py", line 2447, in wsgi_app
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:     response = self.full_dispatch_request()
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:   File "/snap/microk8s/x1/lib/python3.6/site-packages/flask/app.py", line 1952, in full_dispatch_request
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:     rv = self.handle_user_exception(e)
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:   File "/snap/microk8s/x1/lib/python3.6/site-packages/flask/app.py", line 1821, in handle_user_exception
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:     reraise(exc_type, exc_value, tb)
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:   File "/snap/microk8s/x1/lib/python3.6/site-packages/flask/_compat.py", line 39, in reraise
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:     raise value
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:   File "/snap/microk8s/x1/lib/python3.6/site-packages/flask/app.py", line 1950, in full_dispatch_request
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:     rv = self.dispatch_request()
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:   File "/snap/microk8s/x1/lib/python3.6/site-packages/flask/app.py", line 1936, in dispatch_request
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:     return self.view_functions[rule.endpoint](**req.view_args)
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:   File "/snap/microk8s/x1/scripts/cluster/agent.py", line 600, in join_node_dqlite
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:     voters, non_voters = get_dqlite_nodes()
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:   File "/snap/microk8s/x1/scripts/cluster/agent.py", line 469, in get_dqlite_nodes
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]:     data = data.safe_load(f)
Nov 18 10:30:50 u1 microk8s.daemon-cluster-agent[6907]: UnboundLocalError: local variable 'data' referenced before assignment
```

Comment (neoaggelos) [reply scripts/cluster/agent.py]: :facepalm:

**Review by ktsakalozos — APPROVED** [review]
LGTM +1

**Pull request merged**